I3uckwheat/cards-of-carousal
# Feature/223 scoring feature

#### 0. Make sure your issue is linked:

Closes #223 

#### 1. Purpose:

Introduces the score tracking in order to progress with the game and provides a new Game-Over screen to signalize the game is over.


#### 2. Implementation:

Scores are added to the round winner as soon as it is submitted by the host in the `WINNER_SELECTED` event (which is the updated name for the late `SELECT_WINNER`). I created a distinction between `HostRoundWinnerScreen` and `HostGameWinnerScreen` to better represent the current game stage. `HostRoundWinnerScreen` is amended to check whether the `gameSettings.winningScore` is met and if so, a `GAME_OVER` event is dispatched to update the state to the next screen.

The Host is then sent to the new `HostGameWinnerScreen` whilst the players are sent a personalised message depending on whether they won or lost the game.



#### 3. Possible Issues:

This PR is put as a Draft because I didn't manage to finish the test suite for `HostGameWinnerScreen`. Also, I left an important TODO on said screen that states that the "Start new game" button should ideally reset the game to a blank state, but preserve players connections and game settings. This is quite a bit of work that needs attention to work properly but for now the button merely closes the lobby.


#### 4. How to Test:

1. Open multiple instances of the app (3 tabs minimum)
2. Host the game in one tab
3. Join as players with the rest
4. *Set Winning Score to 1*
5. Start the game
6. Play normally
7. See the game over screen as the winning score is met after one round

#### 5. Cleanup
